tests/test_unit/test_utils/tests_external_id_factories.py

```diff
@@ -0,0 +1,215 @@
+import uuid
+from dataclasses import dataclass
+from typing import Callable
+
+import pytest
+
+from cognite.pygen.utils.external_id_factories import (
+    ExternalIdFactory,
+    create_external_id_factory,
+    domain_name_factory,
+    external_id_generator,
+    incremental_factory,
+    sha256_factory,
+    shorten_string,
+    uuid_factory,
+)
+
+
+@dataclass
+class FooWrite:
+    name: str
+
+
+@dataclass
+class FooBarWrite:
+    name: str
+
+
+@dataclass
+class FooBarApply:
+    name: str
+
+
+@pytest.mark.parametrize(
+    "input_str, length, expected",
+    [
+        ("fooBar", 0, ""),
+        ("fooBar", 1, "f"),
+        ("fooBar", 2, "fo"),
+        ("fooBar", 3, "foo"),
+        ("fooBar", 4, "fooB"),
+        ("fooBar", 5, "fooBa"),
+        ("fooBar", 6, "fooBar"),
+        ("fooBar", 7, "fooBar"),
+        ("12", 1, "1"),
+        ("14", 1, "1"),
+    ],
+)
+def test_shorten_string(input_str: str, length: int, expected: str):
+    actual = shorten_string(input_str, length)
+
+    assert actual == expected
+
+
+@pytest.mark.parametrize(
+    "domain_cls, data, shorten_length, expected, short_expected",
+    [
+        (FooWrite, {}, 7, "foo", "foo"),
+        (FooBarWrite, {}, 5, "foobar", "fooba"),
+        (FooBarApply, {}, 7, "foobarapply", "foobara"),
+    ],
+)
+def test_domain_name_factory(domain_cls: type, data: dict, shorten_length: int, expected: str, short_expected: str):
+    actual = domain_name_factory(domain_cls, data)
+    domain_name_factory.shorten_length = shorten_length
+    short_actual = domain_name_factory.short(domain_cls, data)
+
+    assert actual == expected
+    assert short_actual == short_expected
+
+
+@pytest.mark.parametrize(
+    "domain_cls, data, shorten_length",
+    [
+        # TODO: Add more test cases with data
```

Comment:
Why have test cases with data `uuid_factory` is not using any data as far as I understand. Only test I can imagen is a test with and without data to ensure that data is not used by the factory method.
```suggestion
```

Reply:
removed, had just forgotten to clean it up